src/logic/lemmas.lean

```diff
@@ -0,0 +1,46 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import tactic.split_ifs
+
+/-!
+# More basic logic properties
+
+A few more lemmas that could not be in `logic.basic` because of import cycles.
```

Comment:
The import cycle is from `split_ifs`, right? Can we instead prove this without the tactic, like every other lemma in `logic.basic`?

Reply:
I tried to get such a short proof without `split_ifs`, but failed. Do you really care about not using `split_ifs`?

Reply:
It seems silly to create a new file for lemmas which are already very similar to lemmas in `logic.basic` already just because of a slightly longer proof

Reply:
Sure, but it seems silly to make proofs longer because we want to shoehorn lemmas into a given file. Files are cheap, and it's pretty common to have a pair of "very basic"-"higher imports" basic files

Reply:
I think I'm okay with a new file; I don't like unnecessarily avoiding helpful tactics.

But let's just change this sentence, to say "These are in their own file, rather than `logic.basic`, because it is convenient to be able to use the `split_ifs` tactic."

Reply:
> Sure, but it seems silly to make proofs longer because we want to shoehorn lemmas into a given file. Files are cheap, and it's pretty common to have a pair of "very basic"-"higher imports" files.
> 
> I can imagine this new file getting fuller and fuller over time.

In that case, do you want to move lemmas from `logic.basic` here to golf some proofs and get that process started? :)

Reply:
Sure! but I would rather it be dealt with as it comes.